src/main/java/picard/analysis/FastWgsMetricsCollector.java

```diff
@@ -0,0 +1,195 @@
+        this.pileupSize = counterManager.newCounter();
+        this.unfilteredDepthSize = counterManager.newCounter();
+    }
+
+    @Override
+    public void addInfo(final AbstractLocusInfo<EdgingRecordAndOffset> info, final ReferenceSequence ref, boolean referenceBaseN) {
+        prepareCollector(info);
+        for (final EdgingRecordAndOffset record : info.getRecordAndOffsets()) {
+            final String readName = record.getReadName();
+            Optional<Set<EdgingRecordAndOffset>> recordsAndOffsetsForName = Optional.ofNullable((Set<EdgingRecordAndOffset>)readsNames.get(readName));
+            if (record.getType() == EdgingRecordAndOffset.Type.BEGIN) {
+                processRecord(info.getPosition(), ref, record, recordsAndOffsetsForName.orElse(new HashSet<>()));
+            } else {
+                recordsAndOffsetsForName.ifPresent(
+                        edgingRecordAndOffsets -> removeRecordFromMap(record,
+                                edgingRecordAndOffsets));
+            }
+        }
+        if (!referenceBaseN) {
+            final int readNamesSize = pileupSize.get(info.getPosition());
+            final int highQualityDepth = Math.min(readNamesSize, coverageCap);
+            if (highQualityDepth < readNamesSize) {
+                basesExcludedByCapping += readNamesSize - coverageCap;
+            }
+            highQualityDepthHistogramArray[highQualityDepth]++;
+            unfilteredDepthHistogramArray[unfilteredDepthSize.get(info.getPosition())]++;
+        }
+    }
+
+    private void processRecord(int position, ReferenceSequence ref, EdgingRecordAndOffset record, Set<EdgingRecordAndOffset> recordsAndOffsetsForName) {
+        long processedLoci = counter;
+        readsNames.put(record.getReadName(), recordsAndOffsetsForName);
+        final byte[] qualities = record.getBaseQualities();
+        final byte[] bases = record.getRecord().getReadBases();
+        for (int i = 0; i < record.getLength(); i++) {
+            final int index = i + position;
+            if (isReferenceBaseN(index, ref)) {
+                continue;
+            }
+            final byte quality = qualities[i + record.getOffset()];
+            if (quality <= 2) {
+                basesExcludedByBaseq++;
+            } else {
+                if (unfilteredDepthSize.get(index) < coverageCap) {
+                    unfilteredBaseQHistogramArray[quality]++;
+                    unfilteredDepthSize.increment(index);
+                }
+                if (quality < collectWgsMetrics.MINIMUM_BASE_QUALITY || SequenceUtil.isNoCall(bases[i + record.getOffset()])){
+                    basesExcludedByBaseq++;
+                } else {
+                    final int bsq = excludeByQuality(recordsAndOffsetsForName, index);
+                    if (recordsAndOffsetsForName.size() - bsq > 0) {
+                        basesExcludedByOverlap++;
```

Comment:
I"m confused about this. you call `excludeByQuality` and then increment `basesExcludedByOverlap`?